### PR TITLE
Dev doc: Add docs for Trials, SearchSpace, Annotation and GridSearch

### PR DESCRIPTION
**Pull request by Crysple**

### CODE REVIEW

**Comment by Crysple** [comment]
This pr is not ready and still being developed

File: docs/AnnotationSpec.md [hunk 1]
```diff
@@ -4,24 +4,22 @@ For good user experience and reduce user effort, we need to design a good annota
 
 If users use NNI system, they only need to:
 
- 1. Use nni.get_next_parameter() to retrieve hyper parameters from Tuner, before using other annotation, use following annotation at the begining of trial code:
-    '''@nni.get_next_parameter()'''
+ 1. Annotate variables in code as:
```
Comment (QuanluZhang) [inline docs/AnnotationSpec.md]: please give more explanation about usage, for example, there must be an assignment expression below it.

File: docs/AnnotationSpec.md [hunk 2]
```diff
@@ -4,24 +4,22 @@ For good user experience and reduce user effort, we need to design a good annota
...
 
- 2. Annotation variable in code as:
+    `'''@nni.variable(nni.choice(2,3,5,7),name=self.conv_size)'''`
 
-    '''@nni.variable(nni.choice(2,3,5,7),name=self.conv_size)'''
+ 2. Annotate functions in code as:
```
Comment (QuanluZhang) [inline docs/AnnotationSpec.md]: please give more explanation.

File: docs/AnnotationSpec.md [hunk 3]
```diff
@@ -1,58 +1,108 @@
...
- 9. `@nni.variable(nni.lognormal(label, mu, sigma),name=variable)`  
-    Which means the variable value is a value drawn according to exp(normal(mu, sigma))
+ 4. Annotate final result in code as:
+
+    `'''@nni.report_final_result(test_acc)'''`
+
+
+### Types of Search Space
+
+For `@nni.variable`, **`nni.choice`** is the type of search space and there are 10 types to express your search space as follows:
```
Comment (xuehui1991) [inline docs/AnnotationSpec.md]: Shall we out the types of Search Space to another file? Because it not only design for annotation.

**Comment by PurityFan** [comment]
It seems that you haven't made any changes of GridSearch, but you declare it in the title?

File: docs/SearchSpaceSpec.md [hunk 4]
```diff
@@ -67,8 +72,19 @@ The candidate type and value for variable is here:
    * Suitable for a discrete variable with respect to which the objective is smooth and gets smoother with the size of the variable, which is bounded from one side.
 <br/>
 
-Note that SMAC only supports a subset of the types above, including `choice`, `randint`, `uniform`, `loguniform`, `quniform(q=1)`. In the current version, SMAC does not support cascaded search space (i.e., conditional variable in SMAC).
 
-Note that GridSearch Tuner only supports a subset of the types above, including `choic`, `quniform` and `qloguniform`, where q here specifies the number of values that will be sampled. Details about the last two type as follows
+## Relation With Tuners
```
Comment (QuanluZhang) [inline docs/SearchSpaceSpec.md]: looks good. could you change the title to "Search space types supported by each Tuner".
Then, please list all the tuners in the table below.

File: docs/Trials.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+In the yaml configure file, you need to set *useAnnotation* to true to enable NNI annotation:
+```
+useAnnotation: true
+```
+
+## Others
+
+### Logging
+
+In NNI, you can use `logging` module for logging, which will be shown in WebUI (including reported metrics).
```
Comment (QuanluZhang) [inline docs/Trials.md]: did you check with @yds05 about this line?

File: docs/Trials.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+Other outputs will be re-directed to the `stdout` file.
+
+### Advance APIs
+
+#### get_sequence_id
```
Comment (QuanluZhang) [inline docs/Trials.md]: please formally describe this api, including usage, arguments, returns.

**Pull request merged**